near/omni-tests/src/omni_token.rs

```diff
@@ -0,0 +1,361 @@
+                .call("new")
+                .args_json(json!({
+                    "prover_account": prover_contract.id(),
+                    "mpc_signer": "mpc.testnet",
+                    "nonce": U128(0),
+                    "wnear_account_id": "wnear.testnet",
+                }))
+                .max_gas()
+                .transact()
+                .await?
+                .into_result()?;
+
+            // Setup token deployer
+            let token_deployer = worker
+                .create_tla_and_deploy(
+                    account_n(1),
+                    worker.dev_generate().await.1,
+                    &std::fs::read(TOKEN_DEPLOYER_PATH)?,
+                )
+                .await?
+                .unwrap();
+
+            token_deployer
+                .call("new")
+                .args_json(json!({
+                    "controller": locker.id(),
+                    "dao": AccountId::from_str("dao.near").unwrap(),
+                }))
+                .max_gas()
+                .transact()
+                .await?
+                .into_result()?;
+
+            // Configure locker
+            locker
+                .call("add_token_deployer")
+                .args_json(json!({
+                    "chain": init_token_address.get_chain(),
+                    "account_id": token_deployer.id(),
+                }))
+                .max_gas()
+                .transact()
+                .await?
+                .into_result()?;
+
+            locker
+                .call("add_factory")
+                .args_json(json!({
+                    "address": init_token_address,
+                }))
+                .max_gas()
+                .transact()
+                .await?
+                .into_result()?;
+
+            // Deploy token
+            let token_contract =
+                Self::deploy_token(&worker, &locker, &init_token_address, &token_metadata).await?;
+
+            Ok(Self {
+                worker,
+                locker,
+                token_contract,
+                init_token_address,
+                token_metadata,
+            })
+        }
+
+        async fn deploy_token(
+            worker: &near_workspaces::Worker<near_workspaces::network::Sandbox>,
+            locker: &near_workspaces::Contract,
+            init_token_address: &OmniAddress,
+            token_metadata: &BasicMetadata,
+        ) -> anyhow::Result<near_workspaces::Contract> {
+            let token_deploy_initiator = worker
+                .create_tla(account_n(2), worker.dev_generate().await.1)
+                .await?
+                .unwrap();
+
+            let required_storage: NearToken = locker
+                .view("required_balance_for_deploy_token")
+                .await?
+                .json()?;
+
+            token_deploy_initiator
+                .call(locker.id(), "deploy_token")
+                .args_borsh(get_test_deploy_token_args(
+                    ChainKind::Eth,
```

Comment:
Here, ChainKind should match the one used in init_token_address, and it’s not always ETH.

Reply:
And where should it be checked? In the prover?